### DISCUSSION TITLE
Firefly III Data Importer broke down.

### DISCUSSION THREAD

**Discussion by Nchinasss** [firefly-iii/firefly-iii#8097] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

Whoops! 500 :(
Sorry, the Firefly III Data Importer broke down.

Error message
The "state" returned from your server doesn't match the state that was sent.

More information
The error occurred in /var/www/html/app/Http/Controllers/TokenController.php:74.

Please collect more information in the storage/logs directory, where you will find log files. If you're running Docker, use docker logs -f [container]. You can read more about collecting error information [in the FAQ](https://docs.firefly-iii.org/firefly-iii/faq/other/#how-do-i-enable-debug-mode).

Get help on GitHub
You're more than welcome to open a new issue [on GitHub](https://github.com/firefly-iii/firefly-iii/issues).

Use the search!
Include the information [from this debug page](http://www.sqzj.love:9990/debug).
Tell us more than "it says Whoops!"
Include error logs (see above).
Tell us what you were doing.

### Debug information

[2023-10-24 22:57:37] production.DEBUG: Now at App\Http\Controllers\TokenController::callback  
[2023-10-24 22:57:37] production.ERROR: State according to session: ""  
[2023-10-24 22:57:37] production.ERROR: State returned in request : "C3ajZ9Y7EtFSPc4c8MPoKhhl3TvmYE46vaiHpJor"  
[2023-10-24 22:57:37] production.ERROR: The "state" returned from your server doesn't match the state that was sent. {"exception":"[object] (App\\Exceptions\\ImporterErrorException(code: 0): The \"state\" returned from your server doesn't match the state that was sent. at /var/www/html/app/Http/Controllers/TokenController.php:74)
[stacktrace]
#0 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Controller.php(54): App\\Http\\Controllers\\TokenController->callback()
#1 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(43): Illuminate\\Routing\\Controller->callAction()

**Comment by JC5** [comment]
What's the way you run the data importer?

**Reply by Nchinasss** [reply]
docker

**Reply by JC5** [reply]
Are you approaching the importer directly on its IP, by any chance?

**Reply by 2ndwonder** [reply]
I have the same issue, did you guys figure it out?

**Reply by danrasmuson** [reply]
Same issue here

**Reply by danrasmuson** [reply]
Update: I got this working by removing the checkbox "Confidential" as stated in the instructions of the data importer. Its a gotcha and the error message right now could be improved, but good to see how I misconfigured it.

**Reply by s3w47m88** [reply]
Same issue here. But removing Confidential didn't make a difference. Also on Docker, ARM (Apple M3 Pro) in case that matters.

**Reply by Theushmi** [reply]
I'm facing the same issue, configured as per the documentation. Has anyone found a solution?

**Reply by FeltenVanballenberghe** [reply]
Same issue. When authorising the "Authorization Request" for the data importer registration. Running on docker, not the standard ports.

#Trouble Shoot - progress
I've connected the data-importer with a Personal Access Token instead of using the Client way.
It jumps to the data importer dashbord but you can not use the import tasks.
It reports:

`Configuration / connection error :(
The importer could not connect to Firefly III. Please remedy the error below first, and check out the [documentation](https://docs.firefly-iii.org/references/faq/data-importer/general/) if necessary.
Tried "http://xxxxx/api/v1/about?" 5 times but failed: cURL error 28: Failed to connect to xxx port pppp after 31410 ms: Timeout was reached (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for http://xxx:pppp/api/v1/about
`

LOGS:

`2025-08-16 11:23:50] local.DEBUG: Now trying to authenticate with Firefly III at http://xxxx:ppp
2025/08/16 11:23:50 [error] 192#192: *56 open() "/var/www/html/public/build/webfonts/fa-solid-900.woff2" failed (2: No such file or directory), client: xxxxxx, server: _, request: "GET /build/webfonts/fa-solid-900.woff2 HTTP/1.1", host: "xxxx:ppp", referrer: "http://xxxxx:pppp+1/build/assets/app-CMc1M8pK.css"
[2025-08-16 11:24:22] local.NOTICE: Could NOT authenticate with Firefly III at http://xxxx:ppp  
[2025-08-16 11:24:22] local.ERROR: Could not connect to Firefly III: Tried "http://xxxxx:pppp/api/v1/about?" 5 times but failed: cURL error 28: Failed to connect to xxxx port pppp after 31410 ms: Timeout was reached (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for http://xxxx:ppp/api/v1/about  
[2025-08-16 11:24:22] local.DEBUG: Using access token "eyJ0eXAiOiJKV1QiLCJhbGciO" (limited to 25 chars if present)  
xxxxx - - [16/Aug/2025:11:24:22 +0200] "GET /token/validate HTTP/1.1" 200 232 "http://xxxx:pppp+1/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/135.0.0.0 Safari/537.36 OPR/120.0.0.0" "-"
`

**Reply by bennysp** [reply]
Same issue. I am on the latest using Kubernetes. Here is an error in the log. Not sure why "State according to session" is blank...

To add, when it first returns, it is a white page that shows "empty". When I refresh, I have this issue above.

```
[2025-08-15 17:20:49] production.ERROR: State according to session: ""  
[2025-08-15 17:20:49] production.ERROR: State returned in request : "Rb0i1jkjDcoUZQqdaRgnosOAokcaezyYboAiYW7i"  
[2025-08-15 17:20:49] production.ERROR: The "state" returned from your server doesn't match the state that was sent. {"exception":"[object] (App\\Exceptions\\ImporterErrorException(code: 0): The \"state\" returned from your server doesn't match the state that was sent. at /var/www/html/app/Http/Controllers/TokenController.php:78)
[stacktrace]
```

**Reply by bennysp** [reply]
If you guys are okay with it, you can use the the "access token" instead. This allowed me to at least get past the above error. Now, I am doing this on kubernetes, so I just use a Kubernetes Secret and here is my yaml in the event it helps anyone else.

```yaml
importer:
  enabled: true
  config:
    env:
      TZ: "America/Chicago"
  fireflyiii:
    vanityUrl: "https://firefly.somedomain.com"
    auth:
      existingSecret: "accesstoken"
  ingress:
    enabled: true
    className: nginx-app
    annotations:
        cert-manager.io/cluster-issuer: vault-issuer
        cert-manager.io/common-name: firefly-importer.somedomain.com
    hosts:
      - firefly-importer.somedomain.com
    tls:
      - secretName: firefly-importer.somedomain.com-tls
        hosts:
          - firefly-importer.somedomain.com
```

**Reply by FeltenVanballenberghe** [reply]
Access token solved the "authentication registration" error but not the "authentication" error when trying to import.
`local.NOTICE: Could NOT authenticate with Firefly III at http://xxxx:ppp`.

@bennysp importing data works on your stack with PAT?

**Reply by hxhelm** [reply]
I had the same issue where after failing to authenticate the importer, that error page appeared. In my case, I had some DNS issues that were unrelated to firefly (This timeout issue to be exact https://github.com/firefly-iii/firefly-iii/issues/4004).
I just wanted to add that I was able to "reset" the data-importer by clearing the `storage/framework/session` and/or `cache` directory. Generally, there seems to be an issue with the error/session handling that results in a broken data-importer when the authentication process fails.
